app/controllers/api/recipes_controller.rb

```diff
@@ -15,9 +17,28 @@ def update
     @recipe.update(recipe_params)
     render json: { message: 'Your recipe was updated.' }
   end
-  
+
+  def create
+    recipe = current_user.recipes.create(recipe_params)
+    if recipe.persisted?
+      render json: { recipe: recipe, message: 'Your recipe has been created' }, status: 201
+    else
+      render_error(recipe.errors.full_messages.to_sentence, 422)
+    end
+  end
 
   private
+
+  def validate_params_presence
+    if params[:recipe].nil?
+      render_error('Missing params', :unprocessable_entity)
+    elsif params[:recipe][:name].nil?
+      render_error('Your recipe must have a name', :unprocessable_entity)
+    elsif params[:recipe][:instructions].nil?
+      render_error('Your recipe must have instructions', :unprocessable_entity)
+    end
+  end
```

Comment:
All of this (perhaps with the exception of `params[:recipe]` will be handled by the recipe model

Reply:
We tried to use custom messages, but if not needed we can remove them